cmd/chartsvc/handler.go

```diff
@@ -217,11 +217,16 @@ func newChartResponse(c *models.Chart) *apiResponse {
 }
 
 func newChartListResponse(charts []*models.Chart) apiListResponse {
+	// We will keep track of unique digest:chart to avoid duplicates
+	chartDigests := map[string]bool{}
 	cl := apiListResponse{}
 	for _, c := range charts {
-		cl = append(cl, newChartResponse(c))
+		digest := c.ChartVersions[0].Digest
+		if _, ok := chartDigests[digest]; !ok {
```

Comment:
I still think it's useful here to add a comment like "filter out the chart if we've seen the same digest before" just as an added clarification to the comment above